src/main/java/org/gradle/profiler/mutations/ApplyChangeToNativeSourceFileMutator.java

```diff
@@ -1,22 +1,42 @@
 package org.gradle.profiler.mutations;
 
+import org.apache.commons.io.FilenameUtils;
 import org.gradle.profiler.BuildContext;
 
 import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class ApplyChangeToNativeSourceFileMutator extends AbstractFileChangeMutator {
 
+    // Both lists taken from https://gcc.gnu.org/onlinedocs/gcc/Overall-Options.html
+    private final List<String> nativeSourcecodeFileEndings = Arrays.asList(
+        ".c", ".cc", ".cp", ".cxx", ".cpp", ".CPP", ".c++", ".C"
+    );
+
+    private final List<String> nativeHeaderFileEndings = Arrays.asList(
+        ".h", ".hh", ".H", ".hp", ".hxx", ".hpp", ".HPP", ".h++", ".tcc"
+    );
+
     public ApplyChangeToNativeSourceFileMutator(File file) {
         super(file);
-        if (!sourceFile.getName().endsWith(".cpp") && !sourceFile.getName().endsWith(".h") && !sourceFile.getName().endsWith(".hpp")) {
-            throw new IllegalArgumentException("Can only modify C++ source or header files");
+        String fileExtension = "." + FilenameUtils.getExtension(sourceFile.getName());
+        List<String> extensionList = Stream.concat(nativeSourcecodeFileEndings.stream(), nativeHeaderFileEndings.stream())
```

Comment:
```suggestion
        boolean isSupportedExtension = Stream.concat(nativeSourcecodeFileEndings.stream(), nativeHeaderFileEndings.stream()).anyMatch(fileExtension::equals)
```

How about ☝️ , then you don't need to create the extra list.

Reply:
Ahh yeah way better, I usually develop in Kotlin so I wasn't aware of the java way of doing this.